### ISSUE TITLE
Bug:  failed to sync stale SNATs on node (AWS IPI - OKD 4.13)

### ISSUE THREAD

**Issue by dan1el-k** [okd-project/okd#1662]
**Describe the bug**

We have reproducible identified an issue, using the latest OKD 4.13.x version on our TEST/DEV clusters which get daily shutdown and restarted for cost-saving.
For fully online clusters we couldn't reproduce this issue so far using the same versions.

Behaviour:

- Node events log:
```sh
error creating gateway for node x.x.x.x failed to init shared interface gateway: failed to sync stale SNATs on node x.x.x.x: unable to fetch podIPs for pod y
```
- ovnkube-master-x pods restart regularly 
- other pods can't be scheduled because of 
```sh
failed to create pod network sandbox k8s_dns-default-6cmft_openshift-dns_b29b1565-dc51-47eb-a0de-2c2992ca2781_0(f5716c1772d370e108f22a22d7bc153f31582eb6c62db518f0a980b18b6e59e4): error adding pod openshift-dns_dns-default-6cmft to CNI network "multus-cni-network": plugin type="multus" name="multus-cni-network" failed (add): [openshift-dns/dns-default-6cmft/b29b1565-dc51-47eb-a0de-2c2992ca2781:ovn-kubernetes]: error adding container to network "ovn-kubernetes": CNI request failed with status 400: '[openshift-dns/dns-default-6cmft
```

- journald on a worker node gives such error:
```sh
Jul 20 08:32:18 ip-10-223-214-72 kubenswrapper[2402]: E0720 08:32:18.562275    2402 pod_workers.go:965] "Error syncing pod, skipping" err="failed to \"CreatePodSandbox\" for \"kyverno-7c6af-bftgh-981784655_pipeline(33e07ce0-79ba-48f7-b9ac-5adad8457faf)\" with CreatePodSandboxError: \"Failed to create sandbox for pod \\\"kyverno-7c6af-bftgh-981784655_pipeline(33e07ce0-79ba-48f7-b9ac-5adad8457faf)\\\": rpc error: code = Unknown desc = failed to create pod network sandbox k8s_kyverno-7c6af-bftgh-981784655_pipeline_33e07ce0-79ba-48f7-b9ac-5adad8457faf_0(34d9cb3d240e2865b1716d443d2aac67769fc065be0a44985c3f060d6dbac2c2): error adding pod ckyverno-7c6af-bftgh-981784655 to CNI network \\\"multus-cni-network\\\": plugin type=\\\"multus\\\" name=\\\"multus-cni-network\\\" failed (add): [pipeline/kyverno-7c6af-bftgh-981784655/33e07ce0-79ba-48f7-b9ac-5adad8457faf:ovn-kubernetes]: error adding container to network \\\"ovn-kubernetes\\\": CNI request failed with status 400: '[pipeline/yverno-7c6af-bftgh-981784655 34d9cb3d240e2865b1716d443d2aac67769fc065be0a44985c3f060d6dbac2c2] [pipeline/kyverno-7c6af-bftgh-981784655 34d9cb3d240e2865b1716d443d2aac67769fc065be0a44985c3f060d6dbac2c2] failed to get pod annotation: timed out waiting for annotations: context deadline exceeded\\n'\"" pod="pipeline/kyverno-7c6af-bftgh-981784655" podUID=33e07ce0-79ba-48f7-b9ac-5adad8457faf
```

**Version**
Environment: AWS IPI
Tested on OKD version:

- 4.13.0-0.okd-2023-07-09-062029
- 4.13.0-0.okd-2023-06-24-145750
- 4.13.0-0.okd-2023-06-04-080300

**How reproducible**
1. Install a fresh cluster in the versions (mentioned above) / or update to this version.
2. Shutdown the cluster cleanly (first worker nodes, then master nodes).
3. Wakeup the cluster in reverse order (first master nodes, then worker nodes)
4. Checking Node Events and Pods not schedulable

**Additional information**

Perhaps solved by: https://github.com/openshift/ovn-kubernetes/pull/1724 

Seems this issue was introduced by:
https://github.com/openshift/ovn-kubernetes/pull/1662/files and/or https://github.com/openshift/ovn-kubernetes/pull/1715

**Comment by vrutkovs** [comment]
If its OCPBUGS-13175 do we need to track it in OKD specifically?

**Comment by dan1el-k** [comment]
> If its [OCPBUGS-13175](https://issues.redhat.com/browse/OCPBUGS-13175) do we need to track it in OKD specifically?

Just wanted to bring this to attention, as this potentially hold people back from upgrading (like my team).
I didn't find another way how to get support or prioritisation out of the OKD community for the upstream bugfix.

But feel free to close, if you see this as not necessary.

**Comment by vrutkovs** [comment]
Ah, okay. I don't mind keeping it open, just wanted to make sure every single OCP bug is not opened as affecting OKD.

Out of curiousity - for how long do the clusters get shutdown usually? Also, this is baremetal, so it gets powered off, not suspended as a VM, right?

**Comment by dan1el-k** [comment]
> do

Clusters are running on air-gapped AWS accounts. So Nodes are simply shutdown from within one master node (after etc-d backup) and wakeup via AWS API.

They get shutdown in the evening 08.00 p.m and waked up at 06:00 a.m automatically during the week. 
At weekend they don't get waked up. 
This is purely for cost-saving for TEST/DEV clusters

**Comment by dan1el-k** [comment]
> Ah, okay. I don't mind keeping it open, just wanted to make sure every single OCP bug is not opened as affecting OKD.
> 
> Out of curiousity - for how long do the clusters get shutdown usually? Also, this is baremetal, so it gets powered off, not suspended as a VM, right?



> Ah, okay. I don't mind keeping it open, just wanted to make sure every single OCP bug is not opened as affecting OKD.
> 
> Out of curiousity - for how long do the clusters get shutdown usually? Also, this is baremetal, so it gets powered off, not suspended as a VM, right?

rs are running on air-gapped AWS accounts. So Nodes are simply shutdown from within one master node (after etc-d backup) and wakeup via AWS API.

They get shutdown in the evening 08.00 p.m and waked up at 06:00 a.m automatically during the week. 
At weekend they don't get waked up. 
This is purely for cost-saving for TEST/DEV clusters

**Comment by vrutkovs** [comment]
Ah, okay, so its no longer than a few days, that's within kubelet cert expiration time, so it should be fairly straightforward

**Comment by dan1el-k** [comment]
Fixed and tested with 4.13.0-0.okd-2023-08-04-164726
